container-chains/runtime-templates/simple/src/lib.rs

```diff
@@ -1125,6 +1129,37 @@ impl_runtime_apis! {
             SLOT_DURATION
         }
     }
+
+    impl xcm_fee_payment_runtime_api::XcmPaymentApi<Block> for Runtime {
+        fn query_acceptable_payment_assets(xcm_version: staging_xcm::Version) -> Result<Vec<VersionedAssetId>, XcmPaymentApiError> {
+            if !matches!(xcm_version, 3 | 4) {
```

Comment:
here we need to return (if we can) whatever we have in the mapping of assetRate. assetRate defines asset_id to native asset-rates, so we first need to iterate what Ids we have there, then query ForeignAssetCreator to retrieve the ML with respet to an asset_id

Reply:
Reading the keys of `pallet_asset_rate::ConversionRateToNative` I get an `AssetId` which is defined as `u16`, can I use that `u16` to create a `VersionedAssetId` directly? Or do I need the `ForeignAssetCreator`? Because this method doesn't need the multilocation

Reply:
Nevermind, I do need the location